File: sdkwork-app-sdk-typescript/src/types/member-role-update-form.ts
/** 成员角色更新请求 */
export interface MemberRoleUpdateForm {
  /** 工作空间ID */
  workspaceId?: string;
  /** 成员ID */
  memberId?: string;
  /** 用户ID */
  userId?: string;
  /** 新角色ID */
  newRoleId?: string;
  /** 新角色名称 */
  newRoleName?: string;
  /** 权限列表 */
  permissions?: string[];
  /** 更新原因 */
  updateReason?: string;
  /** 是否需要通知 */
  needNotify?: boolean;
}

File: sdkwork-app-sdk-go/types/member_role_update_form.go
package types

// 成员角色更新请求
type MemberRoleUpdateForm struct {
	WorkspaceId string `json:"workspaceId"`
	MemberId string `json:"memberId"`
	UserId string `json:"userId"`
	NewRoleId string `json:"newRoleId"`
	NewRoleName string `json:"newRoleName"`
	Permissions []string `json:"permissions"`
	UpdateReason string `json:"updateReason"`
	NeedNotify bool `json:"needNotify"`
}
